src/app/component/login-page/login-page.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Location } from '@angular/common';
+import { Component } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+import { AuthService } from '../../service/auth.service';
+
+@Component({
+  selector: 'app-login-page',
+  templateUrl: './login-page.component.html',
+  styleUrls: ['./login-page.component.scss']
+})
+export class LoginPageComponent {
+  loginForm = this.fb.group({
+    username: [''],
+    password: ['']
+  })
+
+  constructor(
+    private authService: AuthService,
+    private location: Location,
+    private fb: FormBuilder
+  ) { }
+
+  onSubmit(): void {
+    this.authService.login(this.loginForm.value.username, this.loginForm.value.password).subscribe(
+      success => this.location.back(),
+      error => alert(error)
```

Comment:
you could try to add the validation errors in the html, so it will be a nicer :))